docs/how_to/deploy/adreno.rst

```diff
@@ -0,0 +1,266 @@
+
+   target="opencl -device=adreno"
+
+   with tvm.transform.PassContext(opt_level=3):
+      graph, lib, params = relay.build_module.build(mod, target, params=params)
+   print(lib.imported_modules[0].get_source())
+
+The kernels generated this way is actually working with 2d arrays, leveraging textures
+
+.. code:: c
+
+   __kernel void tvmgen_default_fused_nn_conv2d_kernel0(__write_only image2d_t pad_temp_global_texture, __read_only image2d_t p0) {
+   // body..
+
+Precisions
+~~~~~~~~~~
+
+We can also set different precision, choosing from *float16*,
+*float16_acc32* (Mixed Precision), *float32*. Choosing lower precision may positively
+affect the performance of the model, but it may also have a decrease in the accuracy of the model.
+In some tasks we may sacrifice accuracy in favor of speed, and in some tasks we may prefer mixed precision,
+obtaining some kind of balance.
+
+First of all, to be able to convert precisions in general, we need to
+register conversion to mixed precision
+
+.. code:: python
+
+   from  tvm.relay.op  import  register_mixed_precision_conversion
+
+   conv2d_acc = "float32"
+
+   # Pick a priority > 10 to overwrite defaults, higher priorities take precedence
+   @register_mixed_precision_conversion("nn.conv2d", level=11)
+   def  conv2d_mixed_precision_rule(call_node: "relay.Call", mixed_precision_type: str):
+       global  conv2d_acc
+       return [
+           # always do main calculation in mixed_precision_type
+           relay.transform.mixed_precision.MIXED_PRECISION_ALWAYS,
+           # the dtype for the accumulator
+           conv2d_acc,
+           # the output dtype for the operation (usually fp16)
+           mixed_precision_type,
+       ]
+
+   # Same for dense
+   @register_mixed_precision_conversion("nn.dense", level=11)
+   def  conv2d_mixed_precision_rule(call_node: "relay.Call", mixed_precision_type: str):
+       global  conv2d_acc
+       return [
+           relay.transform.mixed_precision.MIXED_PRECISION_ALWAYS,
+           conv2d_acc,
+           mixed_precision_type,
+       ]
+
+We then need to create a Relay graph from desired model in any convinient way
+and obtain **mod** (which is IR representation of the model), after which we can convert it to
+required **dtype** and then assemble our model sequentialy
+
+.. code:: python
+
+   def  convert_to_dtype(mod, dtype):
+       # downcast to float16
+       if  dtype == "float16"  or  dtype == "float16_acc32":
+           global  conv2d_acc
+           conv2d_acc = "float16"  if  dtype == "float16"  else  "float32"
+           from  tvm.ir  import  IRModule
+           mod = IRModule.from_expr(mod)
+           seq = tvm.transform.Sequential(
+               [
+                   relay.transform.InferType(),
+                   relay.transform.ToMixedPrecision()
+               ]
+           )
+           with  tvm.transform.PassContext(opt_level=3):
+               mod = seq(mod)
+       return  mod
+
+   dtype="float16_acc32"
```

Comment:
I would not like mention mixed precision in the first presentatoin of ToMixedPrecision at all. If we want to cover mixed precisoin, let's describe conversion into fp16 and then have a section with regestring functions, etc, etc